--- a/assemble	2019-08-22 11:09:01.256952569 +0200
+++ b/assemble	2019-08-22 11:10:12.294612866 +0200
@@ -18,20 +18,6 @@
   fi
 }
 
-# Install pipenv to the separate virtualenv to isolate it
-# from system Python packages and packages in the main
-# virtualenv. Executable is simlinked into ~/.local/bin
-# to be accessible. This approach is inspired by pipsi
-# (pip script installer).
-function install_pipenv() {
-  echo "---> Installing pipenv packaging tool ..."
-  VENV_DIR=$HOME/.local/venvs/pipenv
-  virtualenv_bin "$VENV_DIR"
-  $VENV_DIR/bin/pip --isolated install -U pipenv
-  mkdir -p $HOME/.local/bin
-  ln -s $VENV_DIR/bin/pipenv $HOME/.local/bin/pipenv
-}
-
 set -e
 
 shopt -s dotglob
@@ -44,7 +30,6 @@
 fi
 
 if [[ ! -z "$ENABLE_PIPENV" ]]; then
-  install_pipenv
   echo "---> Installing dependencies via pipenv ..."
   if [[ -f Pipfile ]]; then
     pipenv install --deploy
